Storage/tests/System/ManageBucketsTest.php

```diff
@@ -160,29 +160,65 @@ public function testUpdateBucketWithLifecycleDeleteRule(array $rule, $isError =
         $this->assertEquals($lifecycle->toArray(), $bucket->info()['lifecycle']);
     }
 
-    public function testUpdateBucketWithAutoclassConfig()
+    /**
+     * @dataProvider autoclassConfigs
+     */
+    public function testCreateAndUpdateBucketWithAutoclassConfig($autoclassConfig)
     {
-        $autoclassConfig = [
-            'autoclass' => [
-                'enabled' => true,
-            ],
-        ];
+        $autoclassConfig = ['autoclass' => $autoclassConfig];
 
         $bucket = self::createBucket(
             self::$client,
             uniqid(self::TESTING_PREFIX),
             $autoclassConfig
         );
         $this->assertArrayHasKey('autoclass', $bucket->info());
-        $this->assertTrue($bucket->info()['autoclass']['enabled']);
+        $autoclassInfo = $bucket->info()['autoclass'];
+        $this->assertTrue($autoclassInfo['enabled']);
+        $this->assertArrayHasKey('toggleTime', $autoclassInfo);
+        $this->assertArrayHasKey('terminalStorageClass', $autoclassInfo);
+        if (array_key_exists('terminalStorageClass', $autoclassConfig)) {
+            $this->assertEquals(
+                $autoclassConfig['terminalStorageClass'],
+                $autoclassInfo['terminalStorageClass']
+            );
+        }
+        $this->assertArrayHasKey('terminalStorageClassUpdateTime', $autoclassInfo);
 
         // test disabling autoclass
-        $autoclassConfig['autoclass']['enabled'] = false;
+        $autoclassConfig = ['autoclass' => ['enabled' => false]];
         $bucket->update($autoclassConfig);
         $this->assertArrayHasKey('autoclass', $bucket->info());
         $this->assertFalse($bucket->info()['autoclass']['enabled']);
     }
 
+    /**
+     * @dataProvider autoclassConfigs
+     */
+    public function testUpdateExisitngBucketWithAutoclassConfig($autoclassConfig)
```

Comment:
This test seems to fail on prod:
```
Google\Cloud\Storage\Tests\System\ManageBucketsTest::testUpdateExisitngBucketWithAutoclassConfig with data set #2 (array(true, 'ARCHIVE'))
Google\Cloud\Core\Exception\BadRequestException: {"error":{"code":400,"message":"Autoclass is not available for existing buckets in this project: 968036764985","errors":[{"message":"Autoclass is not available for existing buckets in this project: 968036764985","domain":"global","reason":"invalid"}]}}

/usr/local/google/home/vishwarajanand/github/google-cloud-php/Storage/vendor/google/cloud-core/src/RequestWrapper.php:398
/usr/local/google/home/vishwarajanand/github/google-cloud-php/Storage/vendor/google/cloud-core/src/RequestWrapper.php:215
/usr/local/google/home/vishwarajanand/github/google-cloud-php/Storage/vendor/google/cloud-core/src/RestTrait.php:101
/usr/local/google/home/vishwarajanand/github/google-cloud-php/Storage/src/Connection/Rest.php:723
/usr/local/google/home/vishwarajanand/github/google-cloud-php/Storage/src/Connection/Rest.php:210
/usr/local/google/home/vishwarajanand/github/google-cloud-php/Storage/src/Bucket.php:1014
/usr/local/google/home/vishwarajanand/github/google-cloud-php/Storage/tests/System/ManageBucketsTest.php:207

```

Reply:
The feature has been allow listed only for particular testing projects in each library as can be seen [here](https://docs.google.com/document/d/11MqjaM94_mn0pFSLF9LYdbcAQWGw3IKfUhazZrXJSu0/edit?resourcekey=0-Imw-PsXwtQ0e18C9a1drxw#heading=h.en763rrte7qu).

Reply:
Got it.. I can see my tests passing now.. thanks for the pointer.